### ISSUE TITLE
 camera.onNewDing.subscribe doesn't work with motion anymore

### ISSUE THREAD

**Issue by sjiampojamarn** [dgreif/ring#210]
I mostly followed 
[ring/examples/example.ts
](https://github.com/dgreif/ring/blob/master/examples/example.ts)

```
camera.onNewDing.subscribe(ding => {
        const event =
          ding.kind === 'motion'
            ? 'Motion detected'
            : ding.kind === 'ding'
            ? 'Doorbell pressed'
            : `Video started (${ding.kind})`
```

Only ding and other events were detected but not when there were motions. 

It used to work but not now. I'm not sure if it's related to recent changes in F2A/OAuth2 I have seen in HomeAssistant and python-ring-doorbell projects where I got them working with recent fixes. 

However, for this node ring, somehow my motion detection capability's gone while other functions seem to be working just fine. 

Is anyone experiencing the same?   

FYI: my ring git's synced to **HEAD, tag: v5.10.0**
$ node -v
v8.10.0

**Comment by dgreif** [comment]
I'm guessing this is related to #181.  Basically, the polling endpoint from Ring just randomly stops returning motion events for some users.  Can you test with the python library to confirm whether or not it receives motion events right now?  If the python version is receiving motion events, but the node version isn't, then I can dig into their implementation and see if I'm doing anything different.

**Comment by sjiampojamarn** [comment]
yes, the python library has just come back and been working after syncing up to 0.2.8
https://github.com/tchellomello/python-ring-doorbell/pull/159

**Comment by schmupu** [comment]
I have the same problem. I am using ring-client-api in version 5.12.0, node 10.17.0, npm 6.11.13. I set the polling to 20 and 2 seconds like in your example.  With `camera.onNewDing.subscribe(ding => { });` I get only once the while a ring (ding) or motion. Sometimes it does not work for days.

**Comment by ZerGo0** [comment]
Same problem here, worked the past 2 days and now it randomly stopped working, both onNewDing and onMotionDetected.

**Comment by dgreif** [comment]
I would love to get this figured out, but unfortunately I can't reproduce the issue on my end.  Ring has had some really weird issues with the `dings` endpoint over the last few months and I'm guessing this is still an issue on their end.  The fact that it works from the python library but not the node library is what is really bugging me (per @sjiampojamarn).  If any of you are able to debug this and help me figure out what's different between my implementation and the python version, I would really appreciate it!

**Comment by sjiampojamarn** [comment]
Here what I have tried and noticed so far:

- if I ask the Python one to send a video recording request whenever there is a motion to the node one, I can see that the node also get 'motion' event. But it seems that the node one can't get the motion event without any trigger from the Python one. Note that ding (ring the bell) event works fine without any workaround.

- I saw the node declare for v9 in the API but Python uses v11. Not sure if anything to do with it. I tried to just change to v11. No luck there either. 

- I haven't read the code throughout but it seems to me that the Python one calls up for event history --last 50 or something if my memory is right rather than some kind of event pushing. My node skill is even worse :-(. I'll try to dig more.

Anything else I can try or look into specifically.

**Comment by sjiampojamarn** [comment]
> * I saw the node declare for v9 in the API but Python uses v11. Not sure if anything to do with it. I tried to just change to v11. No luck there either.

Got this flipped. Node uses V11 and Python one uses V9.

**Comment by dgreif** [comment]
@sjiampojamarn where do you see the node library registering for version 9 of the api?  Can you give me a file/line number?

Regarding the different ding kinds, please note that there are more than just `ding` and `motion`.  There is also `on_demand` (which you get when you request a new live stream manually), and `alarm`/`on_demain_link` which occur when recording is triggered from other sources.  Does knowing that help you solve your issue at all?

**Comment by sjiampojamarn** [comment]
Yep. I noticed it about on_demand and motion. Node did get on_demand and also split out the motion along as well. It is the same if I ring the bell, both ding and motion get detected in the node no problem but not when there is only motion without a push from Python (on demand).

The version is in ring/api/rest-client.ts
Line 16.

**Comment by dgreif** [comment]
I'm really confused about the app version.  The node client should be on v11, not v9 - https://github.com/dgreif/ring/blob/master/api/rest-client.ts#L16.  Are you using an old fork/version of the node library?

**Comment by sjiampojamarn** [comment]
> I'm really confused about the app version. The node client should be on v11, not v9 - https://github.com/dgreif/ring/blob/master/api/rest-client.ts#L16. Are you using an old fork/version of the node library?

Right, I got it flipped. Node uses V11 and Python one used V9:
https://github.com/tchellomello/python-ring-doorbell/blob/7d8fc425f8c5c3682761f7f7919c6d0860f39e06/ring_doorbell/const.py#L49

I tried to change from V11 to V9 -- just simple one line change there. I'm not sure if anything to do with it or anything else I should change but from there. No luck all looks the same.

**Comment by dgreif** [comment]
I think I found the issue!!  Ring has always had a concept of a `session` based on a `hardware_id`.  What I never realized before was that each session can have its own settings.  In this case, the session can subscribe and unsubscribe to motion and ding events.  This is how each device that has the Ring app (phone, tablet, pc) can activate or deactivate the notifications for ding or motion events.  _Somehow_ your sessions settings must have gotten updated to unsubscribe from the motion events.  I've added logic to automatically re-subscribe to those events for any camera that is unsubscribed when the code starts up.  Please give v`5.13.0` a try and see if you start receiving events again 😄

**Comment by sjiampojamarn** [comment]
> I think I found the issue!! Ring has always had a concept of a `session` based on a `hardware_id`. What I never realized before was that each session can have its own settings. In this case, the session can subscribe and unsubscribe to motion and ding events. This is how each device that has the Ring app (phone, tablet, pc) can activate or deactivate the notifications for ding or motion events. _Somehow_ your sessions settings must have gotten updated to unsubscribe from the motion events. I've added logic to automatically re-subscribe to those events for any camera that is unsubscribed when the code starts up. Please give v`5.13.0` a try and see if you start receiving events again

Yes. Quick check. It works!!. Thank you for the great finding and awesome work.
